crates/fuel-core/src/service/sub_services.rs

```diff
@@ -109,8 +109,23 @@ pub fn init_sub_services(
     };
 
     #[cfg(feature = "p2p")]
-    let p2p_adapter =
-        P2PAdapter::new(network.as_ref().map(|network| network.shared.clone()));
+    let p2p_adapter = {
+        use crate::service::adapters::PeerReportConfig;
+
+        // Hardcoded for now, but left here to be configurable in the future.
+        let peer_report_config = PeerReportConfig {
+            successful_block_import: 5.,
+            missing_block_headers: -100.,
+            bad_block_header: -100.,
+            missing_transactions: -100.,
+            invalid_transactions: -100.,
```

Comment:
This configuration might belong in the consensus parameters. Can we capture this work in a new issue?

Reply:
I don't believe this should go in consensus parameters, as it's specific to a single P2P implementation. But I agree that they could be passed in at the same place as the consensus parameters. 

I'll create an issue.

Reply:
Actually. I think it might still fall into this current issue:
https://github.com/FuelLabs/fuel-core/issues/943

Reply:
https://github.com/FuelLabs/fuel-core/issues/1340